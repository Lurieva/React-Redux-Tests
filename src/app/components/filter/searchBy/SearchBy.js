import React, { Component } from 'react';
import PropTypes from 'prop-types';

import { SEARCH_BY } from '../../../app.config';
import RadioGroup from '../../../shared/RadioGroup/RadioGroup';

import './searchBy.scss';

const SearchBy = ({ searchBy, onChange }) => (
    <div>
        <RadioGroup name={'search'}
                    groupClassName={'searchBy'}
                    param={searchBy} 
                    options={[SEARCH_BY.TITLE, SEARCH_BY.DIRECTOR]}
                    title={'Search By'}
                    onChange={onChange} />
    </div>
);

SearchBy.propTypes = {
    searchBy: PropTypes.string.isRequired,
    onChange: PropTypes.func.isRequired,
};

export default SearchBy;