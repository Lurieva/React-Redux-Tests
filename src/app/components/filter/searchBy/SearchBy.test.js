import React from 'react';
import { shallow } from 'enzyme';
import SearchBy from './SearchBy';

jest.mock('../../../shared/RadioGroup/RadioGroup', () => 'RadioGroup');

describe('SearchBy', () => {
    const mockFn = jest.fn();

    it('should be defined', () => {
        expect(SearchBy).toBeDefined();
    });

    it('should render correctly ', () => {
        const tree = shallow(<SearchBy searchBy='search'
                                       onChange={mockFn} />);
        expect(tree).toMatchSnapshot();
    });
});