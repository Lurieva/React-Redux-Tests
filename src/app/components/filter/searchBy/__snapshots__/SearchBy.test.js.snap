// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBy should render correctly  1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SearchBy
    onChange={[MockFunction]}
    searchBy="search"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "groupClassName": "searchBy",
      "name": "search",
      "onChange": [MockFunction],
      "options": Array [
        Object {
          "label": "Title",
          "value": "title",
        },
        Object {
          "label": "Director",
          "value": "director",
        },
      ],
      "param": "search",
      "title": "Search By",
    },
    "ref": null,
    "rendered": null,
    "type": "RadioGroup",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "groupClassName": "searchBy",
        "name": "search",
        "onChange": [MockFunction],
        "options": Array [
          Object {
            "label": "Title",
            "value": "title",
          },
          Object {
            "label": "Director",
            "value": "director",
          },
        ],
        "param": "search",
        "title": "Search By",
      },
      "ref": null,
      "rendered": null,
      "type": "RadioGroup",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
